### PR TITLE
change behavior when add repo with same name and url.

### LINKED ISSUE #8771: helm repo add now fails in v3.3.2 and v3.3.3
### Description

The command **`helm repo add <name> <url>`** fails as it is no longer idempotent.  This worked fine in version `v3.3.1` and earlier.  It now fails in `v3.3.2` and `v3.3.3`.  This is a breaking change that has broken automation relying on `helm` and tools `helmfile`.

### Steps  to Reproduce

```bash 
helm repo add jaegertracing https://jaegertracing.github.io/helm-charts
helm repo add jaegertracing https://jaegertracing.github.io/helm-charts
```

### Expected Results

```
"jaegertracing" has been added to your repositories
```

I expected that the command would be idempotent and would not fail.  As the repository was already added, it would not need to be added again, and thus should to exist with an error.

### Actual Results

Despite the repository already in the desired state, helm exits with an error, even though there is no error.  `helm` will exit with `1` indicated an error.  This destroys any automation scripts and tools like `helmfile` to function correctly.

```
Error: repository name (jaegertracing) already exists, please specify a different name
```

Output of `helm version`:
```
version.BuildInfo{Version:"v3.3.3", GitCommit:"55e3ca022e40fe200fbc855938995f40b2a68ce0", GitTreeState:"dirty", GoVersion:"go1.15.2"}
```

Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"18", GitVersion:"v1.18.6", GitCommit:"dff82dc0de47299ab66c83c626e08b245ab19037", GitTreeState:"clean", BuildDate:"2020-07-15T23:30:39Z", GoVersion:"go1.14.4", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"17+", GitVersion:"v1.17.9-eks-4c6976", GitCommit:"4c6976793196d70bc5cd29d56ce5440c9473648e", GitTreeState:"clean", BuildDate:"2020-07-17T18:46:04Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}
```

Cloud Provider/Platform (AKS, GKE, Minikube etc.): 

* EKS (K8S `v1.17.9-eks-4c6976`)

### PR DESCRIPTION
**Pull request by yxxhero**
Signed-off-by: yxxhero <aiopsclub@163.com>

**What this PR does / why we need it**:
resolve: https://github.com/helm/helm/issues/8771
We could just return exit code zero if the same repo is attempted to be added twice with the same name.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by joaopaulosr95 — APPROVED** [review]
👍

File: pkg/repo/repo.go [hunk 1]
```diff
@@ -85,6 +85,22 @@ func (r *File) Has(name string) bool {
 	return entry != nil
 }
 
+// HasRepoWithNameAndURL returns true if the given name whth url is already a repository name.
+func (r *File) HasRepoWithNameAndURL(name string, url string) bool {
```
Comment (bacongobbler) [inline pkg/repo/repo.go:89]: Why are we introducing these functions?

File: pkg/repo/repo.go [hunk 2]
```diff
@@ -85,6 +85,22 @@ func (r *File) Has(name string) bool {
...
+	entry := r.GetRepoByNameAndURL(name, url)
+	return entry != nil
+}
+
+// GetRepoByNameAndURL returns an entry with the given name and url if it exists, otherwise returns nil
+func (r *File) GetRepoByNameAndURL(name string, url string) *Entry {
```
Comment (bacongobbler) [inline pkg/repo/repo.go:95]: why is this necessary?

**Review by bacongobbler — CHANGES_REQUESTED** [review]
Thank you for the quick PR. However while reviewing, I do not believe there is need for some of the function calls added here. These newly introduced function calls are overly verbose and are leaky in design.

I will submit an alternative PR to fix this issue. Thank you for your effort.

**Comment by bacongobbler** [comment]
See #8777.
